tests/ops/op_math/test_evolve.py

```diff
@@ -0,0 +1,273 @@
+
+        H = time_dependent_hamiltonian()
+
+        t = jnp.pi / 4
+        params = [1, 2]
+        ev = Evolve(H=H, params=params, t=t)
+
+        time_step = 1e-4
+        times = jnp.arange(0, t, step=time_step)
+        true_mat = jnp.eye(2 ** len(ev.wires))
+        for ti in times[::-1]:
+            true_mat @= qml.math.expm(-1j * time_step * qml.matrix(H(params, ti)))
+
+        assert qml.math.allclose(ev.matrix(), true_mat, atol=1e-2)
+
+
+class TestIntegration:
+    """Integration tests for the ParametrizedEvolution class."""
+
+    # pylint: disable=unused-argument
+    @pytest.mark.jax
+    def test_time_independent_hamiltonian(self):
+        """Test matrix method for a time independent hamiltonian."""
+        import jax
+        from jax import numpy as jnp
+
+        H = time_independent_hamiltonian()
+
+        dev = qml.device("default.qubit", wires=2)
+
+        t = 4
+
+        @qml.qnode(dev, interface="jax")
+        def circuit(params):
+            Evolve(H=H, params=params, t=t, dt=1e-6)
+            return qml.expval(qml.PauliX(0) @ qml.PauliX(1))
+
+        @jax.jit
+        @qml.qnode(dev, interface="jax")
+        def jitted_circuit(params):
+            Evolve(H=H, params=params, t=t, dt=1e-6)
+            return qml.expval(qml.PauliX(0) @ qml.PauliX(1))
+
+        @qml.qnode(dev, interface="jax")
+        def true_circuit(params):
+            true_mat = qml.math.expm(-1j * qml.matrix(H(params, t)) * t)
+            QubitUnitary(U=true_mat, wires=[0, 1])
+            return qml.expval(qml.PauliX(0) @ qml.PauliX(1))
+
+        params = jnp.array([1.0, 2.0])
+
+        assert qml.math.allclose(circuit(params), true_circuit(params), atol=1e-3)
+        assert qml.math.allclose(jitted_circuit(params), true_circuit(params), atol=1e-3)
+        assert qml.math.allclose(
+            jax.grad(circuit)(params), jax.grad(true_circuit)(params), atol=1e-3
+        )
+        assert qml.math.allclose(
+            jax.grad(jitted_circuit)(params), jax.grad(true_circuit)(params), atol=1e-3
+        )
+
+    @pytest.mark.slow
+    @pytest.mark.jax
+    def test_time_dependent_hamiltonian(self):
+        """Test matrix method for a time dependent hamiltonian."""
+        import jax.numpy as jnp
+
+        H = time_dependent_hamiltonian()
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev, interface="jax")
+        def circuit(params, t):
+            Evolve(H=H, params=params, t=t)
+            return qml.expval(qml.PauliZ(0) @ qml.PauliX(1))
+
+        @qml.qnode(dev, interface="jax")
+        def true_circuit(params, t):
+            time_step = 1e-4
+            times = jnp.arange(0, t, step=time_step)
+            true_mat = jnp.eye(2 ** len(H.wires))
+            for ti in times[::-1]:
+                true_mat @= qml.math.expm(-1j * time_step * qml.matrix(H(params, ti)))
+            QubitUnitary(U=true_mat, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0) @ qml.PauliX(1))
```

Comment:
Some potential tricks that might help speed this up so we can keep the gradient test:
  * Use `jax.scipy.linalg.expm`, it is much faster
  * Use `jax.lax.scan` instead of the for loop (this should also automatically jit all the steps

Alternatively, since this is only a 2 qubit system, you can analytically diagonalize the matrix $M = U^\dagger D U$ and use $\exp(-i t M) = U^\dagger \exp(-i t D) U$ (where D is the diagonal matrix the the eigenvalues). Its maybe a bit overkill, but a possibility 😆 

That should help, if it is still unbearably slow (how slow are we talking by the way?) my suggestion is: execute this and its gradient once locally and hardcode the results for the test. Better than not to have the gradient test.

Reply:
If you decide for hardcoding leave a developer comment how the test result was obtained

Reply:
I use `functools.reduce` and `jax.scipy.linalg.expm` and it is much faster now!